contracts/utils/cryptography/SignatureChecker.sol

```diff
@@ -18,14 +18,14 @@ import "../../interfaces/IERC1271.sol";
  */
 library SignatureChecker {
     function isValidSignatureNow(address signer, bytes32 hash, bytes memory signature) internal view returns (bool) {
-        if (Address.isContract(signer)) {
-            try IERC1271(signer).isValidSignature(hash, signature) returns (bytes4 magicValue) {
-                return magicValue == IERC1271(signer).isValidSignature.selector;
-            } catch {
-                return false;
+        if (signature.length == 64 || signature.length == 65) {
```

Comment:
It feels like this length check is not needed, and somehow duplicates code from tryRecover. Maybe we can avoid it

Reply:
I agree, it will cost not much more gas.